src/client/Components/Util/OAuth.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import PropTypes from 'prop-types';
+
+export default class OAuth extends React.Component {
+  state = {
+    user: {},
+    disabled: ''
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.startAuth = this.startAuth.bind(this);
+    this.closeCard = this.closeCard.bind(this);
+  }
+
+  componentDidMount() {
+    const { socket, provider } = this.props;
+
+    socket.on(provider, (user) => {
+      this.popup.close();
+      this.setState({ user });
+    });
+  }
+
+  checkPopup() {
+    const check = setInterval(() => {
+      const { popup } = this;
+      if (!popup || popup.closed || popup.closed === undefined) {
```

Comment:
There has to be something cleaner than having `this.popup` sometimes be defined sometimes be undefined ect. IMO all objects should have some value in their fields at all times. Is there an `Option` type in JS?